src/cmd/flux-terminus.c

```diff
@@ -192,8 +192,10 @@ int main (int argc, char *argv[])
         exit (1);
 
     if ((argc - optindex == 0)
-        || !optparse_get_subcommand (p, argv[optindex]))
+        || !optparse_get_subcommand (p, argv[optindex])) {
         optparse_fatal_usage (p, 1, NULL);
+        exit (1);
```

Comment:
Just curious what's going on here? `optparse_fatal_usage() doesn't return unless the fatalerr function is overridden so the `exit(1)` here *should* be dead code.

Reply:
I believe the fact that the fatalerr function can be overridden to something that doesn't exit is what has GCC tied in knots.  It throws a warning (which gets elevated to an error) about code further down using potentially uninitialized variables.  You probably already saw, but the warning/error message from GCC is in the commit message. 

Rather than adding `exit (1);` here, I wonder if we can add some sort of pragma to the code to tell GCC to ignore any `maybe-uninitialized` warnings emanating from this code.  I feel like disabling `maybe-uninitialized` across the whole code base is too heavy-handed.

Thoughts?

Reply:
Would initializing `id` be a bad idea to get around this?

Reply:
yeah, want to try `__attribute__ ((noreturn))` to `optparse_fatal_usage()`?

Reply:
Though I guess you are right code could override the fatalerr function to actually return so that we can't put that attribute on the function with total confidence... maybe your fix is fine here. Thanks.

Reply:
> Would initializing id be a bad idea to get around this?

Sorry I missed this comment at first. That seems like a pragmatic suggestion @chu11

Reply:
Thanks @grondo and @chu11!  Just replaced the commit with a commit that initializes `id` rather than adding `exit (1)`.